app/domain/authentication/strategy.rb

```diff
@@ -8,15 +8,31 @@ class Strategy < ::Dry::Struct
 
     AuthenticatorNotFound = ::Util::ErrorClass.new(
       "'{0}' wasn't in the available authenticators")
-    ValidationFailed = ::Util::ErrorClass.new(
+    InvalidCredentials = ::Util::ErrorClass.new(
       "Invalid credentials")
 
+
     class Input < ::Dry::Struct
       attribute :authenticator_name, Types::NonEmptyString
       attribute :service_id,         Types::NonEmptyString.optional
       attribute :account,            Types::NonEmptyString
       attribute :username,           Types::NonEmptyString
       attribute :password,           Types::NonEmptyString
+
+      def to_access_request(env)
```

Comment:
Maybe I'm just missing some context, but I don't understand this method name. Looking at the places it's used didn't really help, either. How is it meant to be read?

Reply:
Maybe part of the problem is that I don't really understand what an `Input` is.